xbmc/windows/GUIWindowLoginScreen.cpp

```diff
@@ -279,6 +280,8 @@ void CGUIWindowLoginScreen::LoadProfile(unsigned int profile)
 
   // stop PVR related services
   CServiceBroker::GetPVRManager().Stop();
+  // clear peripherals
+  CServiceBroker::GetPeripherals().Clear();
```

Comment:
I checked the code and it's an error to call Clear() while other systems might still be using the Peripheral Manager. CPeripherals::Clear() should be made private.

Reply:
The reason why I put `Clear()` call here is the `CPeripheral.Initialize()` uses profile specified settings. The idea was reload settings which were loaded on stage two after loading profile.